pylint/config/config_initialization.py

```diff
@@ -147,3 +154,48 @@ def _config_initialization(
             for arg in parsed_args_list
         )
     )
+
+
+def _order_all_first(config_args: list[str], *, joined: bool) -> list[str]:
```

Comment:
Did you consider making this part of `ArgumentsManager`? It feels like we want to make this the default behaviour and don't want it to be possible to parse arguments without this behaviour right?

Reply:
We do so here:
https://github.com/pylint-dev/pylint/blob/f815e9f5fcdd5caffc04e335fc4590a65989ef21/pylint/pyreverse/main.py#L279
Although `all` isn't used here.

I also think we might skip in some test setups.

I'm not sure, we do quite a bit of parsing in `config_initialization` as well and perhaps `ArgumentsManager` should be kept small. Let's see what Pierre thinks.

Reply:
I'm also not very clear on the design if the argument parser, I usually have simple argparse add_argument calls with custom type if required in my code. The current design might be due to optparse being used before or the need to pre-parse some if the arguments ? Anyway, it seems to work as is, so I don't think committing too much in the current design or a refactor of the current design or this merge request would be worth it.